wpilibj/src/main/java/edu/wpi/first/wpilibj/command/FunctionalCommand.java

```diff
@@ -0,0 +1,89 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.command;
+
+/**
+ * A FunctionalCommand runs a given Runnable when executed.
+ */
+public class FunctionalCommand extends InstantCommand {
+  // The operation to run when the command is run.
+  private final Runnable m_func;
+
+  /**
+   * Create a command that depends on the given subsystem and calls the provided
+   * func when run.
+   *
+   * @param func       The func to take when the command is run.
+   */
+  public FunctionalCommand(Runnable func) {
+    if (func == null) {
+      throw new IllegalArgumentException("Function must not be null.");
+    }
+    m_func = func;
+  }
+
+  /**
+   * Create a command that depends on the given subsystem and calls the provided
+   * func when run.
+   *
+   * @param subsystem The subsystem that the command depends on.
+   * @param func      The func to take when the command is run.
+   */
+  public FunctionalCommand(Subsystem subsystem, Runnable func) {
+    requires(subsystem);
+    if (func == null) {
+      throw new IllegalArgumentException("Function must not be null.");
+    }
+    m_func = func;
+  }
+
+  /**
+   * Create a named command that depends on the given subsystem and calls the
+   * provided func when run.
+   *
+   * @param name      The name of the command.
+   * @param func      The func to take when the command is run.
+   */
+  public FunctionalCommand(String name, Runnable func) {
+    super(name);
+    if (func == null) {
+      throw new IllegalArgumentException("Function must not be null.");
+    }
+    m_func = func;
+  }
+
+  /**
+   * Create a named command that depends on the given subsystem and calls the
+   * provided func when run.
+   *
+   * @param name      The name of the command.
+   * @param subsystem The subsystem that the command depends on.
+   * @param func      The func to take when the command is run.
+   */
+  public FunctionalCommand(String name, Subsystem subsystem, Runnable func) {
+    super(name);
+    requires(subsystem);
+    if (func == null) {
+      throw new IllegalArgumentException("Function must not be null.");
+    }
+    m_func = func;
+  }
+
+  /**
+   * Trigger the stored func.
+   *
+   * <p>Called just before this Command runs the first time.
+   */
+  @Override
+  protected void _initialize() {
+    if (m_func != null) {
```

Comment:
The constructor bails out on null, so this if statement will always evaluate to true.